charts/opentelemetry-collector/templates/service.yaml

```diff
@@ -12,6 +12,9 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.service.type }}
+  {{- if .Values.service.headless }}
+  clusterIP: "None"
+  {{- end}}
```

Comment:
What if user wants to specify a custom `clusterIP` for the service? Can we make it more flexible? Probably `clusterIP` option instead of `headless` boolean?

Reply:
Hey @dmitryax!  @the-bets and @kurumoorthi89 working on my team but yeah we can totally make it a string field instead of a boolean. As for why we chose the boolean over the string. I honestly didn't think there was a use case for specifying a `clusterIP` other than "None" for headless or "" to let the cluster auto assign an IP.

Reply:
I also never run into that use-case but this option exists https://kubernetes.io/docs/concepts/services-networking/service/#choosing-your-own-ip-address , so I don't think we should restrict it by the chart

Reply:
sounds good. Also we're working out some stuff on the CLA front as an fyi (long story)

Reply:
@dmitryax I have changed the configuration as per the requirement. Now we can able to define headless service or define custom clusterIP.

Please note we're still waiting on CLA stuff. Shall keep posted on the same.